lib/executor/per_vu_iterations.go

```diff
@@ -194,11 +194,30 @@ func (pvi PerVUIterations) Run(ctx context.Context, out chan<- stats.SampleConta
 	defer activeVUs.Wait()
 
 	regDurationDone := regDurationCtx.Done()
-	runIteration := getIterationRunner(pvi.executionState, pvi.logger, out)
-
-	handleVU := func(vu lib.VU) {
+	runIteration := getIterationRunner(pvi.executionState, pvi.logger)
+
+	vusDone := make(chan struct{})
+	activeVUsCount := new(uint64)
+	// Wait for all VUs to finish
+	defer func() {
+		for i := uint64(0); i < atomic.LoadUint64(activeVUsCount); i++ {
+			<-vusDone
+		}
+	}()
```

Comment:
This seems wrong. You'd atomically load `activeVUsCount` on every loop cycle, which doesn't make sense, considering that it's static and known - isn't it basically `numVUs`. What's more, isn't the `activeVUs` wait group already doing all of this?

Can't you just move the `activeVUs.Done()` out of the `defer` and into `DeactivateCallback` and delete all of this duplicate logic? `VUActivationParams.Done` doesn't really seem necessary here.

Reply:
Yeah, I was trying to get rid of the `activeVUs` WG in some cases in favor of this `Done` channel, but it didn't work out for all executors.

I'll give a try using the `DeactivateCallback` for this, which should clean it up as you say.

Reply:
This should be resolved in the latest push.